test/helper_test.exs

```diff
@@ -0,0 +1,126 @@
+                 ]
+               ],
+               queue: queue_name
+             })
+
+    assert :ok = Queue.unbind(meta[:chan], queue_name, exchange_name)
+    assert :ok = Queue.unbind(meta[:chan], queue_name_errored, exchange_name_errored)
+    assert :ok = Exchange.delete(meta[:chan], exchange_name)
+    assert :ok = Exchange.delete(meta[:chan], exchange_name_errored)
+    assert {:ok, %{message_count: 0}} = Queue.delete(meta[:chan], queue_name)
+    assert {:ok, %{message_count: 0}} = Queue.delete(meta[:chan], queue_name_errored)
+  end
+
+  test "configuration without an exchange and with routing key set with correct dead letter queue should not raise an error",
+       meta do
+    queue_name = rand_name()
+    routing_key_name = rand_name()
+    exchange_name = rand_name()
+
+    queue_name_errored = "#{queue_name}_errored"
+
+    assert :ok =
+             Helper.declare(meta[:chan], %{
+               exchanges: [
+                 %{name: exchange_name, opts: [durable: true], routing_keys: [routing_key_name], type: :topic}
+               ],
+               opts: [
+                 durable: true,
+                 arguments: [
+                   {"x-dead-letter-exchange", :longstr, ""},
+                   {"x-dead-letter-routing-key", :longstr, queue_name_errored}
+                 ]
+               ],
+               queue: queue_name
+             })
+
+    assert :ok = Queue.unbind(meta[:chan], queue_name, exchange_name)
+    assert :ok = Exchange.delete(meta[:chan], exchange_name)
+    assert {:ok, %{message_count: 0}} = Queue.delete(meta[:chan], queue_name)
+    assert {:ok, %{message_count: 0}} = Queue.delete(meta[:chan], queue_name_errored)
+  end
+
+  test "bad configuration with dead letter exchange empty and routing key set should rise an error ", meta do
+    queue_name = rand_name()
+    routing_key_name = rand_name()
+    exchange_name = rand_name()
+
+    queue_name_errored = "BadDeadLetterQueue"
+
+    assert_raise RuntimeError,
+                 "Configuring x-dead-letter-exchange with empty string, x-dead-letter-routing-key should be '#{queue_name}_errored' instead of '#{queue_name_errored}'",
+                 fn ->
+                   Helper.declare(meta[:chan], %{
+                     exchanges: [
+                       %{name: exchange_name, opts: [durable: true], routing_keys: [routing_key_name], type: :topic}
+                     ],
+                     opts: [
+                       durable: true,
+                       arguments: [
+                         {"x-dead-letter-exchange", :longstr, ""},
+                         {"x-dead-letter-routing-key", :longstr, queue_name_errored}
+                       ]
+                     ],
+                     queue: queue_name
+                   })
+                 end
+  end
+
+  test "bad configuration with dead letter exchange and routing key should rise an error ", meta do
```

Comment:
```suggestion
  test "bad configuration with dead letter exchange and routing key should raise an error", meta do
```

Reply:
Mi sembra che questa descrizione manchi un `empty` o qualcosa del genere i.e. `bad configuration with empty dead letter exchange and routing key should raise an error`, visto che appunto testa che siano entrambe vuote